pandas/tests/arrays/test_datetimelike.py

```diff
@@ -57,6 +57,54 @@ def timedelta_index(request):
 
 class TestDatetimeArray(object):
 
+    def test_array_object_dtype(self, tz_naive_fixture):
```

Comment:
It think it would be good to add such a test to the base extension tests as well?

Reply:
I don't know those tests well enough to have an informed opinion.  AFAIK ExtensionArray doesn't implement `__array__`, so it isn't clear that this is supported in the general case.

Reply:
EA implements `__iter__`, which should be sufficient.

This test would be slightly opinionated for a base test, in case an EA wants to be converted to a specific NumPy type, but I think it's OK.

Reply:
I guess we have `base.interface.BaseInterfaceTests.test_array_interface` which checks

```python
    def test_array_interface(self, data):
        result = np.array(data)
        assert result[0] == data[0]
```

Reply:
Ah, yes, that's already a generic test. OK, since that does not actually test the return dtype, it's good to have more explicit tests here.

Should we expect from EA that `np.array(EA, dtype=object)` always works (returns an object array of scalars)? 
That seems like an OK assumption to me, since this already happens if you don't implement `__array__`, so we can expect this as well *if* the EA author implements a custom `__array__` I think.